### PR TITLE
Update error msg for passwd-protected PDF

### PR DESCRIPTION
**Pull request by catileptic**
* change public-facing error message to something more explicit
* log other types of errors that might arise

### CODE REVIEW

File: ingestors/documents/pdf.py [hunk 1]
```diff
@@ -20,8 +28,13 @@ def ingest(self, file_path, entity):
         """Ingestor implementation."""
         try:
             self.parse_and_ingest(file_path, entity, self.manager)
+        except pikepdf._qpdf.PasswordError as pwe:
+            raise ProcessingException(
+                "Could not extract PDF file. The PDF is protected with a password. Try removing the password protection and re-ingesting the documents."
+            )
         except Exception as ex:
-            raise ProcessingException("Could not extract PDF file: %r" % ex) from ex
+            log.info(f"Failed to ingest PDF: {file_path}. Reason: {ex}.")
```
Comment (stchris) [inline ingestors/documents/pdf.py:36]: You could use the [exception](https://docs.python.org/3/library/logging.html#logging.Logger.exception) method on the logger, which will print the stack trace as well.

File: ingestors/documents/pdf.py [hunk 2]
```diff
@@ -20,8 +28,13 @@ def ingest(self, file_path, entity):
...
+            raise ProcessingException("Could not extract PDF file.")
```
Comment (stchris) [inline ingestors/documents/pdf.py:37]: Did you intentionally remove the `from ex` from the end of the line here? I think it's generally a good idea to pass the exception up the stack.

**Review by stchris — CHANGES_REQUESTED** [review]

**Review by stchris — COMMENTED** [review]
One more minor comment

**Review by stchris — APPROVED** [review]
Woohoo, thanks for fixing this!

File: ingestors/documents/pdf.py [hunk 3]
```diff
@@ -20,8 +28,13 @@ def ingest(self, file_path, entity):
         """Ingestor implementation."""
         try:
             self.parse_and_ingest(file_path, entity, self.manager)
+        except pikepdf._qpdf.PasswordError as pwe:
+            raise ProcessingException(
+                "Could not extract PDF file. The PDF is protected with a password. Try removing the password protection and re-ingesting the documents."
```
Comment (tillprochaska) [inline ingestors/documents/pdf.py:33]: As this error message will be surfaced to end users, maybe we could use "reupload" instead of "reingest"?

**Pull request closed without merging**